tests/suites/test_suite_psa_crypto.data

```diff
@@ -1161,7 +1161,7 @@ depends_on:MBEDTLS_ARC4_C:MBEDTLS_CIPHER_MODE_CTR
 cipher_setup:PSA_KEY_TYPE_ARC4:"000102030405060708090a0b0c0d0e0f":PSA_ALG_CTR:PSA_ERROR_NOT_SUPPORTED
 
 PSA cipher: bad order function calls
-depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC
+depends_on:MBEDTLS_AES_C:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_CIPHER_PADDING_PKCS7
```

Comment:
As the dependency is in the code and not in the test case data it is better to add the dependency on the test case function.

Reply:
That's true for the entire dependency list, no?

Reply:
Yes (same as https://github.com/ARMmbed/mbedtls/pull/4012 did for `hash_bad_order` and `mac_bad_order`).